docs/aws/audit/codebuildmonitoring/rules/codedeploy_ec2_minimum_healthy_hosts_configured_remediation.mdx

```diff
@@ -1,7 +1,293 @@
+3. **Create a Python Script**: Write a script to interact with the AWS CodeBuild service.
+4. **Implement the Script**: Use the Boto3 client to fetch the current project configuration and update it with the desired minimum healthy hosts fleet percentage.
+
+This script ensures that the minimum healthy hosts fleet percentage is maintained for your CodeBuild project, preventing misconfigurations.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the CodeBuild service.
+2. In the navigation pane, select "Build projects". This will display a list of all your build projects.
+3. Select the build project you want to check. This will open the project details page.
+4. In the project details page, navigate to the "Environment" section. Here, you can see the "Compute" configuration. If the "Compute type" is set to "Fleet", you can see the "Minimum healthy hosts" setting. This setting should be maintained at a certain percentage to ensure the health of your EC2 instances. If it's not, it indicates a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can use the `describe-projects` command to list all the CodeBuild projects in your AWS account. The command is as follows:
+
+   ```
+   aws codebuild describe-projects
+   ```
+
+3. After getting the list of projects, you can use the `batch-get-projects` command to get detailed information about each project. You need to provide the names of the projects as input to this command. The command is as follows:
+
+   ```
+   aws codebuild batch-get-projects --names project-name1 project-name2
+   ```
+
+4. In the output of the above command, look for the `computeType` field under the `environment` section. If the `computeType` is set to `BUILD_GENERAL1_SMALL`, it means that the EC2 instance type used for the CodeBuild project is small and may not maintain the minimum healthy hosts fleet percentage. 
+
+Please note that AWS CodeBuild does not directly provide a configuration to maintain a minimum healthy hosts fleet percentage. You may need to manage this at the EC2 Auto Scaling group level or use other AWS services like Elastic Beanstalk or ECS that provide such configurations.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>